scrapy/utils/retry.py

```diff
@@ -0,0 +1,54 @@
+from scrapy.utils.python import global_object_name
+
+
+class RetryHandler(object):
```

Comment:
This handler does not required any state, so it could be a set of functions.
At least, it could be instantiated only once to store the settings and then the requests could be arguments to methods.